uast/utils.go

```diff
@@ -66,13 +66,64 @@ func CyclomaticComplexity(n *Node)  int {
 		}
 		n := p.Node()
 		for _, r := range n.Roles {
-			switch(r) {
+			switch r {
 			case If, SwitchCase, For, ForEach, While,
-			     DoWhile, TryCatch, Continue, OpBooleanAnd, OpBooleanOr,
-			     OpBooleanXor:
+				DoWhile, TryCatch, Continue, OpBooleanAnd, OpBooleanOr,
+				OpBooleanXor:
 				complx++
 			}
 		}
 	}
 	return complx
 }
+
+func (n *Node) ContainsRole(role Role) bool {
+	for _, r := range n.Roles {
+		if role == r {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *Node) ChildrenOfRole(role Role) []*Node {
```

Comment:
Please move all of these to a function in the npath implementation at the moment.
We are still planning a query API (like the one in the `uast/ann` package) to use for these cases.